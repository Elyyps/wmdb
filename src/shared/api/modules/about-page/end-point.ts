import { generateAboutPageDummyData } from "./generate-dummy-data";

export const generateAboutPageData = generateAboutPageDummyData;
