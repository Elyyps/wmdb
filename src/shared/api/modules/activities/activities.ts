import { IActivityCard } from "@app/api/core/activity-card";

export interface IActivitiesModule {
  items: IActivityCard[];
  title: string;
}
