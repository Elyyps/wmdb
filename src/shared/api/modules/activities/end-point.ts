import { generateActivitiesDummyData } from "./dummy-data";

export const generateActivitiesData = generateActivitiesDummyData;
