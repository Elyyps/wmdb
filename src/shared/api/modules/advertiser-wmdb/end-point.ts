import { generateAdvertiserDummyData } from "./dummy-data";
export const generateAdvertiserData = generateAdvertiserDummyData;
