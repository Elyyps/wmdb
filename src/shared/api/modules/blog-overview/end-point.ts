import { generateBlogOverviewDummyData, generateCards } from "./dummy-data";
import { ICompanyOutingCard } from "@app/api/core/company-outing-card";

export const generateBlogOverviewData = generateBlogOverviewDummyData;
const amountOfCards = 50;
export const getCardsWithPagination = (skip: number, take: number): ICompanyOutingCard[] => {
  let cards: ICompanyOutingCard[] = generateCards(amountOfCards);
  cards = cards.slice(skip, take + skip);

  return cards;
};
