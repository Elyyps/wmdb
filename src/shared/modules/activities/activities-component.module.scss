.activities {
  background-color: #e8f2fc;
  height: 688px;

  h1 {
    padding: 30px 0;
    text-align: center;
  }
  > div {
    width: 80%;
    margin: 0 auto 16px auto;
  }
}
