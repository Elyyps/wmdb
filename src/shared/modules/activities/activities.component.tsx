import * as React from "react";
import styles from "./activities-component.module.scss";
import { IActivitiesModule } from "@app/api/modules/activities/activities";
import { IActivityCard } from "@app/api/core/activity-card";
import { ActivityCardComponent } from "@app/core/activity-card";

export interface IActivitiesComponentProps {
  activitiesModule: IActivitiesModule;
}

const ActivitiesComponent = ({ activitiesModule }: IActivitiesComponentProps) => (
  <div className={styles["activities"]}>
    <h1>{activitiesModule.title}</h1>
    {activitiesModule.items.map((item: IActivityCard, key) => (
      <div key={key} style={{ height: "58px" }}>
        <ActivityCardComponent {...item} key={key} />
      </div>
    ))}
  </div>
);

export { ActivitiesComponent };
