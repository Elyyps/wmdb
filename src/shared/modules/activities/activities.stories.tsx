import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { ActivitiesComponent } from "./activities.component";
import { generateActivitiesData } from "@app/api/modules/activities/end-point";

storiesOf("Activities", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <ActivitiesComponent activitiesModule={generateActivitiesData()} />);
