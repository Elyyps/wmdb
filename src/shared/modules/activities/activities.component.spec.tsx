import React from "react";
import { shallow } from "enzyme";
import { ActivitiesComponent } from "./activities.component";
import { generateActivitiesData } from "@app/api/modules/activities/end-point";

describe("[Activities]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<ActivitiesComponent activitiesModule={generateActivitiesData()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
