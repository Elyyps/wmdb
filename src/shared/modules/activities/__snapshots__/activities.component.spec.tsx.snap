// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Activities] should render component without crashing 1`] = `
<div
  className="activities"
>
  <h1>
    Bekijk ook andere activiteiten!
  </h1>
  <div
    key="0"
    style={
      Object {
        "height": "58px",
      }
    }
  >
    <ActivityCardComponent
      color="#b0cf6d"
      icon="eten.svg"
      key="0"
      title="Eten, Drinken & Feesten"
      total={500}
    />
  </div>
  <div
    key="1"
    style={
      Object {
        "height": "58px",
      }
    }
  >
    <ActivityCardComponent
      color="#71abde"
      icon="teambuilding.svg"
      key="1"
      title="Teambuilden"
      total={500}
    />
  </div>
  <div
    key="2"
    style={
      Object {
        "height": "58px",
      }
    }
  >
    <ActivityCardComponent
      color="#f8b272"
      icon="incentive.svg"
      key="2"
      title="Incentives"
      total={500}
    />
  </div>
  <div
    key="3"
    style={
      Object {
        "height": "58px",
      }
    }
  >
    <ActivityCardComponent
      color="#ef7478"
      icon="spel.svg"
      key="3"
      title="Spel & Ontspanning"
      total={500}
    />
  </div>
  <div
    key="4"
    style={
      Object {
        "height": "58px",
      }
    }
  >
    <ActivityCardComponent
      color="#d6b6d8"
      icon="sportief.svg"
      key="4"
      title="Sportief & Actie(f)"
      total={500}
    />
  </div>
  <div
    key="5"
    style={
      Object {
        "height": "58px",
      }
    }
  >
    <ActivityCardComponent
      color="#75c5c3"
      icon="creatief.svg"
      key="5"
      title="Creatieve Uitjes"
      total={500}
    />
  </div>
  <div
    key="6"
    style={
      Object {
        "height": "58px",
      }
    }
  >
    <ActivityCardComponent
      color="#ffdf7a"
      icon="incentive.svg"
      key="6"
      title="Workshops"
      total={500}
    />
  </div>
</div>
`;
